### PR TITLE
Use native uint128 on Windows with clang-cl

### PR DESCRIPTION
**Pull request by franziskuskiefer**

### CODE REVIEW

**Comment by msprotz** [comment]
thanks for the patch!

I think you might need to change
```
#if !defined(_MSC_VER)
```

into

```
#if !defined(_MSC_VER) || defined(__clang__)
```

a few lines below...?

**Comment by franziskuskiefer** [comment]
So this should be correct now but my Visual Studio 2017 clang-cl crashes when compiling with this. I'll try 2019 later. But this should probably get some testing CI as well 😬

File: include/kremlin/internal/types.h [hunk 1]
```diff
@@ -72,13 +75,13 @@ typedef FStar_UInt128_uint128 FStar_UInt128_t, uint128_t;
...
-#if !defined(_MSC_VER)
+#if !defined(_MSC_VER) || defined(__clang__)
 #include "fstar_uint128_gcc64.h"
 #endif
 
-#if !defined(KRML_VERIFIED_UINT128) && defined(_MSC_VER)
+#if !defined(KRML_VERIFIED_UINT128) && defined(_MSC_VER) && !defined(__clang__)
 #include "fstar_uint128_msvc.h"
-#elif defined(KRML_VERIFIED_UINT128)
+#elif defined(KRML_VERIFIED_UINT128) && !defined(__clang__)
```
Comment (msprotz) [inline include/kremlin/internal/types.h]: I think this one shouldn't be here. 

You still want to enable clang64/windows and verified uint128. This line would make this scenario impossible (I believe).

**Comment by msprotz** [comment]
This looks almost perfect. There's one nit that I've flagged -- once you've fixed it, please go ahead and merge. Thanks for a much appreciated patch!

**Comment by franziskuskiefer** [comment]
This required another small change but should be good now.
Note that `fstar_uint128_gcc64.h` might not be the best name anymore now. It's more like `fstar_uint128_native.h`.

**Pull request merged**